build.gradle

```diff
@@ -0,0 +1,311 @@
+import org.apache.tools.ant.filters.ReplaceTokens
+
+plugins {
+    // github
+    id "co.riiid.gradle" version "0.4.2"
+    // bintray
+    id "com.jfrog.bintray" version "1.6"
+}
+
+// common settings
+subprojects {
+  apply plugin: 'java'
+  apply plugin: 'jacoco'
+  // apply plugin: 'checkstyle'
+  apply plugin: 'findbugs'
+  apply plugin: 'co.riiid.gradle'
+  apply plugin: 'com.jfrog.bintray'
+  apply plugin: 'maven'
+
+  sourceCompatibility = jdkVersion
+  targetCompatibility = jdkVersion
+  [compileJava, compileTestJava]*.options*.encoding = 'UTF-8'
+  [compileJava, compileTestJava]*.options*.bootClasspath = "$JDK_HOME/jre/lib/rt.jar"
+
+  // checkstyle {
+  //  config = resources.text.fromFile("${rootProject.projectDir}/config/checkstyle/checkstyle.xml")
+  //  ignoreFailures = true
+  //  toolVersion = '6.16.1'
+  // }
+
+  findbugs {
+    ignoreFailures = true
+  }
+
+  // Force prevent checkstyle/findbugs on Test.
+  // task checkstyleTest(overwrite: true) << {}
+  task findbugsTest(overwrite: true) << {}
+  tasks.withType(FindBugs) {
+    reports {
+      xml.enabled = false
+      html.enabled = true
+    }
+  }
+
+  repositories {
+    jcenter()
+  }
+
+  dependencies {
+    testCompile 'org.testng:testng:6.9.10'
+  }
+
+  test {
+    useTestNG()
+  }
+
+  javadoc {
+    options.locale = 'en_US'
+  }
+
+  task sourceJar(type: Jar) {
+      from sourceSets.main.allSource
+      classifier = 'sources'
+  }
+
+  task javadocJar(type: Jar, dependsOn: javadoc) {
+      classifier = 'javadoc'
+      from javadoc.destinationDir
+  }
+  artifacts {
+      archives jar
+      archives sourceJar
+      archives javadocJar
```

Comment:
artifacts, jar sourceJar and javadocJar, are not necessary for webbook war project.
It is better to define for necessary projects only.